File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/chainofresponsibility/LevelTwoSupport.java
package chainofresponsibility;

class LevelTwoSupport implements SupportLevel {
    private SupportLevel next;

    @Override
    public void setNext(SupportLevel next) {
        this.next = next;
    }

    @Override
    public void handleRequest(SupportTicket ticket) {
        System.out.println("Level Two Support handling ticket: " + ticket.getDescription());
    }
}

File: code/csharp/DesignPatterns/chainofresponsibility/LevelTwoSupport.cs
namespace ChainOfResponsibility;

class LevelTwoSupport : ISupportLevel
{
    private ISupportLevel _next;

    public void SetNext(ISupportLevel next)
    {
        _next = next;
        Console.WriteLine($"Setting next level to {next
                                                    .GetType().Name} Support");
    }

    public void HandleRequest(SupportTicket ticket)
    {
        Console.WriteLine($"Level Two Support handling ticket: {ticket
                                                                .Description}");
        Console.WriteLine("Ticket resolved.");
    }
}
